### DISCUSSION TITLE
Can I call into ScriptKit script directly from AppleScript

### DISCUSSION THREAD

**Discussion by nikivdev** [johnlindquist/kit#1028] [General]
I would like to call into ScriptKit script without searching for it or using hotkey but do it from Karabiner.

Is it possible to do?

**Comment by johnlindquist** [comment]
@nikitavoloboev Hi Nikita! 👋

Yes, of course! One of the first things I built :)

You have 3 choices:

## 1. Use `~/.kit/kar` to invoke the app (if your script has a prompt)

Lovingly named "Kar" after "Karabiner" ❤️:

```bash
`~/.kit/kar my-script-name
```

This works over a Unix socket. You can also pass args:

```
~/.kit/kar my-script-name arg-1 arg-2
```

## 2. `~/.kenv/bin/my-script-name` for "Terminal" scripts

Each script you create also generates a wrapper in `~/.kenv/bin/my-script-name`. These allow you to interactively run the scripts from the terminal. You can use `arg`, but you won't be able to use app-specific methods/prompts like `editor()`

```
~/.kenv/bin/my-script-name arg-1 arg-2
```

## 3. Write to `~/.kit/run.txt`

I need a "cross-platform" way since Windows doesn't have a Unix Socket. Also for supporting the VS Code extension that let's you run scripts from code: https://marketplace.visualstudio.com/items?itemName=johnlindquist.kit-extension

If you have a script named "say-hi", you can write it to a file called `~/.kit/run.txt`
```
echo "say-hi" > ~/.kit/run.txt
```

Kit.app watches for changes in that file and will run the script when it detects a change. There's no benefit to using this over `kar` for mac users.


## TL:DR; Gimme Karabiner:

I know I don't have to explain this to you 😇

```bash
{:des "kit-mode"
         :rules [:kit-mode
                 [:-q ["~/.kit/kar kit-discussions"]]
                 [:-c ["~/.kenv/bin/code-project ~/.kit"]]
                 [:-h ["echo say-hi > ~/.kit/run.txt"]]
```

**Reply by nikivdev** [reply]
Perfect. Thank you. ♥️
